builder/simplify.cc

```diff
@@ -371,10 +399,16 @@ class simplifier : public node_mutator {
       return;
     }
 
-    // First, prune any dead lets, and any lets that are just vars
+    // - Prune dead lets
+    // - Inline single-ref lets outside of a loop, along with lets that are trivial
     for (auto it = lets.begin(); it != lets.end();) {
-      int refs = *references[it->first];
-      if (refs == 0 || is_variable(it->second, it->first)) {
+      auto& ref_info = references[it->first];
+      assert(ref_info);
+      if (ref_info->ref_count == 0) {
+        it = lets.erase(it);
```

Comment:
I'd remove the two comments on line 402-403, and instead add a comment here:
```
// This let is dead
```
And another one below:
```
// This let is used once, or is simple enough to substitute freely
```

(Same thing for `let` above.)